utils/gear-replay-cli/src/lib.rs

```diff
@@ -16,103 +16,262 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Replaying a block against the live chain state
-
-use clap::{Parser, Subcommand};
-use cmd::*;
 use runtime_primitives::Block;
-use sc_tracing::logging::LoggerBuilder;
-use sp_runtime::traits::{Block as BlockT, Header as HeaderT};
-use std::fmt::Debug;
+#[cfg(feature = "always-wasm")]
+use sc_executor::sp_wasm_interface::HostFunctions;
+#[cfg(not(feature = "always-wasm"))]
+use sc_executor::{HeapAllocStrategy, NativeElseWasmExecutor, NativeExecutionDispatch};
+use sc_executor::{WasmExecutionMethod, WasmExecutor, WasmtimeInstantiationStrategy};
+use sp_core::{
+    offchain::{
+        testing::{TestOffchainExt, TestTransactionPoolExt},
+        OffchainDbExt, OffchainWorkerExt, TransactionPoolExt,
+    },
+    traits::{CallContext, CodeExecutor},
+};
+use sp_externalities::Extensions;
+use sp_keystore::{testing::MemoryKeystore, KeystoreExt, KeystorePtr};
+use sp_rpc::{list::ListOrValue, number::NumberOrHex};
+use sp_runtime::{
+    generic::SignedBlock,
+    traits::{Block as BlockT, HashingFor, Header as HeaderT},
+    DeserializeOwned,
+};
+use sp_state_machine::{
+    backend::BackendRuntimeCode, OverlayedChanges, StateMachine, TestExternalities,
+};
+use std::{fmt::Debug, sync::Arc};
+use substrate_rpc_client::{ChainApi, WsClient};
 
-pub(crate) const LOG_TARGET: &str = "gear_replay";
+pub const LOG_TARGET: &str = "gear_replay";
 
-mod cmd;
+pub mod cmd;
 mod parse;
-mod util;
-
-const VARA_SS58_PREFIX: u8 = 137;
-const GEAR_SS58_PREFIX: u8 = 42;
+mod shared_parameters;
+mod state;
 
-pub(crate) type HashFor<B> = <B as BlockT>::Hash;
-pub(crate) type NumberFor<B> = <<B as BlockT>::Header as HeaderT>::Number;
+pub type HashFor<B> = <B as BlockT>::Hash;
+pub type NumberFor<B> = <<B as BlockT>::Header as HeaderT>::Number;
 
 #[derive(Clone, Debug)]
-pub(crate) enum BlockHashOrNumber<B: BlockT> {
+pub enum BlockHashOrNumber<B: BlockT> {
     Hash(HashFor<B>),
     Number(NumberFor<B>),
 }
 
-/// Commands of `gear-replay` CLI
-#[derive(Debug, Subcommand)]
-pub enum Command {
-    ReplayBlock(replay_block::ReplayBlockCmd<Block>),
-    GearRun(gear_run::GearRunCmd<Block>),
+impl<B: BlockT> core::fmt::Display for BlockHashOrNumber<B> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
```

Comment:
```suggestion
impl<B: BlockT> fmt::Display for BlockHashOrNumber<B> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

and use `std::fmt::{self, Debug};`